modules/data_warehouse/main.tf

```diff
@@ -504,25 +502,79 @@ resource "google_cloudfunctions2_function" "function" {
     service_account_email = google_service_account.cloud_function_service_account.email
   }
 
-  event_trigger {
-    trigger_region = var.region
-    event_type     = "google.cloud.storage.object.v1.finalized"
-    event_filters {
-      attribute = "bucket"
-      value     = google_storage_bucket.provisioning_bucket.name
-    }
-    retry_policy = "RETRY_POLICY_RETRY"
-  }
+  depends_on = [
+    time_sleep.wait_30_seconds
+  ]
+}
+
+# Set up Workflows service account
+# # Set up the Workflows service account
+resource "google_service_account" "workflow_service_account" {
+  project      = module.project-services.project_id
+  account_id   = "cloud-workflow-sa-${random_id.id.hex}"
+  display_name = "Service Account for Cloud Workflows"
+}
+
+# # Grant the Workflow service account Workflows Admin
+resource "google_project_iam_member" "workflow_service_account_admin_role" {
+  project = module.project-services.project_id
+  role    = "roles/workflows.admin"
+  member  = "serviceAccount:${google_service_account.workflow_service_account.email}"
 
   depends_on = [
-    time_sleep.wait_for_eventarc
+    google_service_account.workflow_service_account
   ]
 }
 
-resource "google_project_iam_member" "workflow_event_receiver" {
+# # Grant the Workflow service account Cloud Run Invoke
+resource "google_project_iam_member" "workflow_service_account_invoke_role" {
   project = module.project-services.project_id
-  role    = "roles/eventarc.eventReceiver"
-  member  = "serviceAccount:${data.google_project.project.number}-compute@developer.gserviceaccount.com"
+  role    = "roles/run.invoker"
+  member  = "serviceAccount:${google_service_account.workflow_service_account.email}"
+
+  depends_on = [
+    google_service_account.workflow_service_account
+  ]
+}
+
+# # Grant the Workflow service account ability to run as other account
+resource "google_project_iam_member" "workflow_service_account_act_as_role" {
```

Comment:
We should scope this to just the SA it needs access to using `google_service_account_iam_member`